torch/_subclasses/fake_tensor.py

```diff
@@ -139,15 +138,17 @@ def tree_flatten_only(ty: Type[T], pytree: PyTree):
 # structure. Like `MetaConverter`, it uses `WeakTensorRefKey` to
 # hold a weak reference for all memoized tensors.
 class FakeTensorConverter(object):
-    tensor_memo: weakref.WeakValueDictionary
+    @property
+    def tensor_memo(self):
+        return self.meta_converter.tensor_memo
+
     meta_converter: MetaConverter
     constant_storage_mapping: Dict[StorageWeakRef, List[TensorWeakRef]]
 
     def __init__(self):
-        # FakeTensors store the FakeTensorMode which in turn stores a
-        # FakeTensor, so we need to hold a weak reference to the FakeTensor
-        # otherwise we would induce a circular reference
-        self.tensor_memo = weakref.WeakValueDictionary()
+        # In principle preserving views should be OK, but in practice
+        # AOTAutograd (or maybe autograd) seems to do the wrong thing.  See
+        # https://github.com/pytorch/torchdynamo/issues/1815
```

Comment:
Is this comment out of date now? https://github.com/pytorch/torchdynamo/issues/1815

Reply:
yes, out of date